test/saithrift/tests/saitunnel.py

```diff
@@ -0,0 +1,326 @@
+# (C) Copyright @ 2016 Broadcom.  All Rights Reserved.  The term Broadcom refers to Broadcom Limited and/or its subsidiaries.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Thrift SAI interface Tunnel tests
+"""
+import socket
+
+from switch import *
+
+import sai_base_test
+
+@group("tunnel")
+class IpIpEncapTest(sai_base_test.ThriftInterfaceDataPlane):
+    '''
+    performs encapsulation 
+    we send normal ip packet on port2 and expected to receive encapsulated packet on port 1
+    Underlay route 10.10.1.x
+    inner packet dest ip:1.1.1.1 
+    outer packet dest ip:10.10.1.1
+    -----------------------------------------------------------------
+    Ingress side[port2]           |          Egress side[port1]
+    ------------------------------------------------------------------
+    ip's falls in 20.20.1.0       |        ip's falls in 10.10.1.0
+    ------------------------------------------------------------------
+    TTL MODE : SAI_TUNNEL_TTL_PIPE_MODEL
+    ##Ingress packet=Ether(dst=router_mac)/IP(src=20.20.1.x,dst=1.1.1.1)/TCP()
+    ##Expected encap packet = Ether(src=router_mac,dst=00:05:06:00:00:00)/IP(src=20.20.1.1,dst=1.1.1.1)/IP(src=2.2.2.2,dst=10.10.1.1)/TCP()
+    '''
+    def runTest(self):
+        print
+        switch_init(self.client)
+        port1=port_list[0] #From where we expect encapsulated packet 
+        port2=port_list[1] #From where we send normal ip packet
+        mac1 = '00:05:06:00:00:00' # port1 l2 entry

```

Comment:
mac1 or mac2 are not used, please remove. Can you remove all unnecessary variables?